src/solvers/dg_unstructured_quad/curved_map_and_geom.jl

```diff
@@ -0,0 +1,149 @@
+  Xcomp[4,1], Xcomp[4,2] = evaluate_at( eta, GammaCurves[4] )
+
+  x = ( 0.5 * (  (1.0 - xi)  * Xcomp[4,1] + (1.0 + xi)  * Xcomp[2,1]
+               + (1.0 - eta) * Xcomp[1,1] + (1.0 + eta) * Xcomp[3,1] )
+       - 0.25 * (  (1.0 - xi) * ( (1.0 - eta) * Xref[1,1] + (1.0 + eta) * Xref[4,1] )
+                 + (1.0 + xi) * ( (1.0 - eta) * Xref[2,1] + (1.0 + eta) * Xref[3,1] ) ) )
+
+  y = ( 0.5 * (  (1.0 - xi)  * Xcomp[4,2] + (1.0 + xi)  * Xcomp[2,2]
+               + (1.0 - eta) * Xcomp[1,2] + (1.0 + eta) * Xcomp[3,2] )
+       - 0.25 * (  (1.0 - xi) * ( (1.0 - eta) * Xref[1,2] + (1.0 + eta) * Xref[4,2] )
+                 + (1.0 + xi) * ( (1.0 - eta) * Xref[2,2] + (1.0 + eta) * Xref[3,2] ) ) )
+
+  return x, y
+end
+
+
+# Compute the metric terms for the general curved sided quadrilateral transfitie mapping
+#     Alg. 99 from the blue book of Kopriva
+function transfinite_quad_map_metrics(xi, eta, GammaCurves::Array{GammaCurve,1})
+
+  Xref   = zeros(4,2)
+  Xcomp  = zeros(4,2)
+  Xpcomp = zeros(4,2)
+
+  Xref[1,1], Xref[1,2] = evaluate_at( -1.0, GammaCurves[1] )
+  Xref[2,1], Xref[2,2] = evaluate_at(  1.0, GammaCurves[1] )
+  Xref[3,1], Xref[3,2] = evaluate_at(  1.0, GammaCurves[3] )
+  Xref[4,1], Xref[4,2] = evaluate_at( -1.0, GammaCurves[3] )
+
+  Xcomp[1,1], Xcomp[1,2] = evaluate_at( xi , GammaCurves[1] )
+  Xcomp[2,1], Xcomp[2,2] = evaluate_at( eta, GammaCurves[2] )
+  Xcomp[3,1], Xcomp[3,2] = evaluate_at( xi , GammaCurves[3] )
+  Xcomp[4,1], Xcomp[4,2] = evaluate_at( eta, GammaCurves[4] )
+
+  Xpcomp[1,1], Xpcomp[1,2] = derivative_at( xi , GammaCurves[1] )
+  Xpcomp[2,1], Xpcomp[2,2] = derivative_at( eta, GammaCurves[2] )
+  Xpcomp[3,1], Xpcomp[3,2] = derivative_at( xi , GammaCurves[3] )
+  Xpcomp[4,1], Xpcomp[4,2] = derivative_at( eta, GammaCurves[4] )
+
+  X_xi  = ( 0.5 * (Xcomp[2,1] - Xcomp[4,1] + (1.0 - eta) * Xpcomp[1,1] + (1.0 + eta) * Xpcomp[3,1])
+          -0.25 * ((1.0 - eta) * (Xref[2,1] - Xref[1,1]) + (1.0 + eta) * (Xref[3,1] - Xref[4,1])) )
+
+  X_eta = ( 0.5  * ((1.0 - xi) * Xpcomp[4,1] + (1.0 + xi) * Xpcomp[2,1] + Xcomp[3,1] - Xcomp[1,1])
+           -0.25 * ((1.0 - xi) * (Xref[4,1] - Xref[1,1]) + (1.0 + xi) * (Xref[3,1] - Xref[2,1])) )
+
+  Y_xi = ( 0.5  * (Xcomp[2,2] - Xcomp[4,2] + (1.0 - eta) * Xpcomp[1,2] + (1.0 + eta) * Xpcomp[3,2])
+          -0.25 * ((1.0 - eta) * (Xref[2,2] - Xref[1,2]) + (1.0 + eta) * (Xref[3,2] - Xref[4,2])) )
+
+  Y_eta = ( 0.5  * ((1.0 - xi) * Xpcomp[4,2] + (1.0 + xi) * Xpcomp[2,2] + Xcomp[3,2] - Xcomp[1,2])
+           -0.25 * ((1.0 - xi) * (Xref[4,2] - Xref[1,2]) + (1.0 + xi) * (Xref[3,2] - Xref[2,2])) )
+
+  return X_xi, X_eta, Y_xi, Y_eta
+end
+
+
+# construct the (x,y) node coordinates in the volume of a curved sided element
+function calc_node_coordinates!(node_coordinates, element_id, nodes, poly_deg,
+                                GammaCurves::Array{GammaCurve,1})
+
+  nnodes = poly_deg + 1
+  for j in 1:nnodes, i in 1:nnodes
```

Comment:
```suggestion
  for j in eachindex(nodes), i in eachindex(nodes)
```
This avoids subtle bugs like `for i in nnodes` that we had some time ago... Should probably also be used elsewhere